source/common/router/config_impl.cc

```diff
@@ -513,6 +517,14 @@ void RouteEntryImplBase::finalizeRequestHeaders(Http::HeaderMap& headers,
   vhost_.globalRouteConfig().requestHeaderParser().evaluateHeaders(headers, stream_info);
   if (!host_rewrite_.empty()) {
     headers.Host()->value(host_rewrite_);
+  } else if (auto_host_rewrite_header_) {
+    Http::HeaderEntry* header = headers.get(*auto_host_rewrite_header_);
+    if (header != nullptr) {
+      absl::string_view headerValue = header->value().getStringView();
+      if (!headerValue.empty()) {
+        headers.Host()->value(std::string(headerValue));
```

Comment:
You can leave out the temporary std::string and just pass `headerValue` directly to
 `HeaderEntry::value(absl::string_view)` which will copy the string.

Reply:
Fixed.